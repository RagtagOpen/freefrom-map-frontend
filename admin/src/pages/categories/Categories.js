import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

// Redux
import { getCategoryData, selectCategories } from 'store/categorySlice';

function Categories() {
    const state = useSelector(selectCategories);
    const dispatch = useDispatch();

    useEffect(() => {
        if (state.loaded === false) {
            dispatch(getCategoryData());
        }
    });

    function renderCategory(category, index) {
        return (
            <div className="card" key={`category-${index}`}>
                <div className="card-header" id={`heading-${index}`}>
                    <h5 className="mb-0 d-flex justify-content-between align-items-center">
                        <button
                            className="btn btn-link collapsed"
                            data-toggle="collapse"
                            data-target={`#collapse-${index}`}
                            aria-expanded="false"
                            aria-controls={`collapse-${index}`}>
                            { category.name }
                        </button>
                        <div className="justify-content-end">
                            { category.active ?
                                <span className="badge badge-success badge-pill">Active</span> :
                                <span className="badge badge-danger badge-pill">Inactive</span> }
                            <a className="btn btn-sm btn-primary ml-4" href={`/categories/${category.id}`}>Edit</a>
                        </div>

                    </h5>
                </div>
                <div
                    id={`collapse-${index}`}
                    className="collapse"
                    aria-labelledby={`heading-${index}`}
                    data-parent="#accordion">
                    <div className="card-body">
                        <ul>
                            { category.items.map((item, itemIndex) =>
                                <li key={`category-${index}-${itemIndex}`}>{ item.name }</li>) }
                        </ul>
                    </div>
                </div>
            </div>
        )
    }

    return (
        <div>
            <h1>Scorecard</h1>
            <div id="accordion">
                { state.data.map((category, index) => renderCategory(category, index)) }
            </div>
            <hr className="pb-0" />
            <div className="d-flex align-items-end justify-content-end">
                <a href="/categories/new" title="Create new category" className="btn btn-primary">Create</a>
            </div>
        </div>
    );
}

export default Categories;
