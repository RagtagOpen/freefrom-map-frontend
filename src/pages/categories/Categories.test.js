import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Categories from './Categories';

const mockStore = configureStore([]);

describe('Categories Component', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            categories: {
                loading: false,
                loaded: true,
                data: [
                    {
                        "name": "Test Name",
                        "items": [
                            {"name": "Test Item"}
                        ],
                        "active": true,
                    }
                ]
            }
        });

        component = renderer.create(
            <Provider store={ store }>
                <Categories />
            </Provider>
        );
    });

    it('should match snapshot', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })
})
