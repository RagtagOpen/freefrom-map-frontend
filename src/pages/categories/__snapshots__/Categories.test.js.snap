// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Component should match snapshot 1`] = `
<div>
  <h1>
    Scorecard
  </h1>
  <div
    id="accordion"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
        id="heading-0"
      >
        <h5
          className="mb-0 d-flex justify-content-between align-items-center"
        >
          <button
            aria-controls="collapse-0"
            aria-expanded="false"
            className="btn btn-link collapsed"
            data-target="#collapse-0"
            data-toggle="collapse"
          >
            Test Name
          </button>
          <div
            className="justify-content-end"
          >
            <span
              className="badge badge-success badge-pill"
            >
              Active
            </span>
            <a
              className="btn btn-sm btn-primary ml-4"
              href="/categories/1"
            >
              Edit
            </a>
          </div>
        </h5>
      </div>
      <div
        aria-labelledby="heading-0"
        className="collapse"
        data-parent="#accordion"
        id="collapse-0"
      >
        <div
          className="card-body"
        >
          <ul>
            <li>
              Test Item
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="pb-0"
  />
  <div
    className="d-flex align-items-end justify-content-end"
  >
    <a
      className="btn btn-primary"
      href="/categories/new"
      title="Create new category"
    >
      Create
    </a>
  </div>
</div>
`;
